src/native/libs/System.Native/pal_networkstatistics.c

```diff
@@ -479,29 +492,56 @@ int32_t SystemNative_GetActiveUdpListeners(IPEndPointInfo* infos, int32_t* infoC
         }
         estimatedSize = tmpEstimatedSize;
     }
-    int32_t count = (int32_t)(estimatedSize / sizeof(struct xtcpcb));
+    int32_t count = (int32_t)(estimatedSize / sizeof(struct xinpcb));
```

Comment:
Was xtcpcb wrong?  Or xinpcb is also correct and on more systems?  I'm trying to understand if there's an existing bug here being fixed on supported platforms.

Reply:
yes, `xtcpcb`  described TCP stats - there is more state so it is bigger.

Reply:
we _could_ turn count to (u)long on all platform - I see no harm of using bigger type. This part is common with macOS but as @rootwyrm mentioned unlikely to to have as many UDP servers.

Reply:
Yeah, the socket limit should keep it sane as long as it's filtered by `sysctl`. It's anywhere we intend to walk the entirety that it becomes a very big problem and will need OS-specificity. `xinpcb` contains not only TCP and UDP but also SCTP and raw, so `kptr_t` becomes much more necessary.
I don't think we should change to a `long` here though; TCP/UDP is still defined as 65535 possible ports maximum. If we're seeing more than 65535, something else has probably gone wrong. ;)